Shared/mods/deathmatch/logic/luadefs/CLuaCryptDefs.cpp

```diff
@@ -294,6 +294,57 @@ int CLuaCryptDefs::EncodeString(lua_State* luaVM)
                 }
                 return 1;
             }
+            case StringEncryptFunction::AES128:
+            {
+                SString& key = options["key"];
+                SString& iv = options["iv"];
```

Comment:
I dont think `iv` should be a required field. Actually I'm not even sure if taking user input is a good idea. Rather we should return the hashed data + iv, and let the user store it.
`iv` is [initialization vector](https://www.cryptopp.com/wiki/Initialization_Vector).

Reply:
Maybe you're right, but personally, it would be more convenient for me to generate iv myself. In this case, for example, I can generate and store a key and a iv in advance, in order to then encrypt the files later with the already existing key and iv.

Reply:
Also we can write example function for generating a random string on the wiki, for the convenience of users.

Reply:
We currently do not provide any CSPRNG via the MTA Lua API, and `math.random` can't be assumed suitable for cryptographic purposes, so properly generating IVs in resources is definitely not easy. Also, IVs should **never** be reused. I agree with @Pirulax suggestion here; letting MTA take care of generating a IV is a good idea.

Reply:
Also, what's the use case for knowing the IV in advance? What does that allow you to do that you can't already do after MTA returns the IV it used for ecrnyption?

Reply:
Generally IV is regarded as public information (as per wiki).
So you just storing it somewhere is perfectly fine.
Also, some cipher modes require unique IVs (see wiki I linked above).